src/shell.js

```diff
@@ -43,6 +43,10 @@ var Module = typeof {{{ EXPORT_NAME }}} !== 'undefined' ? {{{ EXPORT_NAME }}} :
 #include "promise_polyfill.js"
 #endif
 
+#if ENVIRONMENT_MAY_BE_SHELL
```

Comment:
OK, how about an internal settings called `INCLUDE_XHR_POLYFILL` which can be set when the file packager is used?  Then we can avoid this unless is needed?